aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -418,9 +415,6 @@ def prepare_process(self):
         if self.ctx.restart_calc:
             self.ctx.inputs.parameters['CONTROL']['restart_mode'] = 'restart'
             self.ctx.inputs.parent_folder = self.ctx.restart_calc.outputs.remote_folder
-        else:
-            self.ctx.inputs.parameters['CONTROL']['restart_mode'] = 'from_scratch'
-            self.ctx.inputs.pop('parent_folder', None)
```

Comment:
I think we should keep this login. The idea is that if there is not restart calculation, we are not restarting, and so we should probably set `from_scratch`. What would happen now if a user provides a `parent_folder` (so wanting do a restart) but sets the incorrect `restart_mode` in the parameters? I am not sure what the behavior will be and maybe QE will internally ignore it and do the right thing, but the idea was to make sure the correct parameter was set. Or maybe it is not even set at all. It would be good to know what happens when the `restart_mode` is inconsistent with what we are requesting. Depending on that, we might want to add this logic to a port validator instead that warns the user in case of inconsistent inputs. But ideally we still do this automatically, if it is acceptable to always override user input in certain cases.

Reply:
I think a validator is indeed the way to go here. If a user wants to restart from _only_ the charge density, the provided input for `startingpot` is ignored if `restart_mode` is set to `'restart'`. So we shouldn't automatically set this value and override the user provided inputs.